jwql/jwql_monitors/generate_preview_images.py

```diff
@@ -212,6 +212,11 @@ def check_existence(file_list, outdir):
         # for the appropriately named jpg of the mosaic, which depends
         # on the specific detectors in the file_list
         file_parts = filename_parser(file_list[0])
+
+        # If filename_parser() does not recognize the filename, return False
+        if not file_parts['recognized_filename']:
```

Comment:
This code (combined with the above code at line 214) implies that if you have an invalid file, you will still get a dictionary back, it's just that the dictionary will have a "recognized_filename" key that is, presumably, a boolean value, with True for a recognized file and False for a non-recognized file.

Reply:
Also, because I forgot, we need to log a warning or error at this point so that we know what happened, and can look through the logs to find it, and to know if we need to update the parsing code.

Reply:
Just a note to confirm that in offline conversations, everyone is now happy with the logging statements in the edited code.